language/move-stdlib/sources/vector.move

```diff
@@ -146,7 +146,25 @@ module std::vector {
         pragma intrinsic = true;
     }
 
-    /// Swap the `i`th element of the vector `v` with the last element and then pop the element.
+    /// Insert `e` at position `i` in the vector `v`, shifting the old `v[i]` and all subsequent elements to the right.
+    /// This is O(n) and preserves ordering of elements in the vector.
+    /// Aborts if `i` is out of bounds.
+    public fun insert<Element>(v: &mut vector<Element>, e: Element, i: u64) {
```

Comment:
I would have expected this to be native. Wondering what the gas cost will be for this swap technique, and whether it is acceptable? @vgao1996